server/webapp/WEB-INF/rails/webpack/views/components/modal/modal_manager.tsx

```diff
@@ -56,17 +56,36 @@ export namespace ModalManager {
     allModals.push({key: modal.id, value: modal});
     document.body.classList.add(styles.fixed);
     m.redraw();
+    addOutsideClickListener(modal);
   }
 
   export function close(modal: Modal) {
     _.remove(allModals, (entry) => entry.key === modal.id);
     document.body.classList.remove(styles.fixed);
     m.redraw();
+    removeOutsideClickListener(modal);
   }
 
   export function closeAll() {
     _.forEach(allModals, (m: ModalEntry) => {
       ModalManager.close(m.value);
     });
   }
+
+  function overlayBackgroundClicked(modal: Modal) {
+    return (event: Event) => {
+      const overlayBg = document.getElementsByClassName(styles.overlayBg).item(0);
+      if (event.target === overlayBg) {
+        modal.close();
+      }
+    };
+  }
+
+  function addOutsideClickListener(modal: Modal) {
+    modalContainer.addEventListener("click", overlayBackgroundClicked(modal));
+  }
+
+  function removeOutsideClickListener(modal: Modal) {
+    modalContainer.removeEventListener("click", overlayBackgroundClicked(modal));
```

Comment:
this will likely cause "leaks" because each invocation of `overlayBackgroundClicked` returns a different function. So technically a listener you add is not the same as the one you remove.

It's likely that the leaked listeners end up doing nothing, because the modal is closed. However they are leaking listeners, and possibly holding a copy of the modal instance itself (along with anything else that the modal hold onto)

Reply:
Ok, I'll look into it and fix it